Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -100,13 +106,51 @@ public BlackboardArtifactNode(BlackboardArtifact artifact) {
         this.setIconBaseWithExtension(ExtractedContent.getIconFilePath(artifact.getArtifactTypeID())); //NON-NLS
     }
 
+    @Override
+    @NbBundle.Messages({"BlackboardArtifactNode.getAction.errorTitle=Error getting actions",
+        "BlackboardArtifactNode.getAction.errorMessage=There was a problem getting actions for the selected result."})
+    public Action[] getActions(boolean context) {
+        List<Action> actionsList = new ArrayList<>();
+        actionsList.addAll(Arrays.asList(super.getActions(context)));
+
+        //see if this artifact has a timestamp in any of the supported attributes
+        boolean hasTimeStamp = false;
+
+        for (ArtifactEventType artEventType : ArtifactEventType.getAllArtifactEventTypes()) {
+            if (artEventType.getArtifactTypeID() == artifact.getArtifactTypeID()) {
+                try {
+                    if (null != artifact.getAttribute(artEventType.getDateTimeAttrubuteType())) {
+                        hasTimeStamp = true;
+                        break;
+                    }
+                } catch (TskCoreException ex) {
+                    LOGGER.log(Level.SEVERE, "Error retreiving blackboard arttributes from blackboard artifact.", ex); //NON-NLS
+                    MessageNotifyUtil.Notify.error(BlackboardArtifactNode_getAction_errorTitle(), BlackboardArtifactNode_getAction_errorMessage());
+                }
+            }
+        }
+        if (hasTimeStamp) {
```

Comment:
This flag could be eliminated by adding the actions and breaking out of the loop.